### PR TITLE
Birthday

### PR DESCRIPTION
**Pull request by nathanleiby**
Issue: https://clever.atlassian.net/browse/INFRA-2082

This introduces a number of re-usable CI scripts, aiming to re-implemented the functionality we got before from Drone plugins. (notably, so far, the `publish` plugins: https://github.com/Clever/drone/tree/master/plugin/publish)

The are implemented to be used with `circleci` (sometimes using env vars that circle CI auto-injects), though they could be written to be more generic if desired.

3 of these are working:

- docker (builds and pushes an image)
- npm (runs npm publish, with some extra error handling)
- report-card (runs report-card)

2 aren't yet implemented:

- gh-release
- catapult

The interface for the plugins is described in the README.

---

My main questions/concerns are:

- do we think the interface to these is stable? if not, do we think it's simple enough to gitbot changes to relevant repos when needed?
- is there any easier way to setup the required environment? Because of how circleCI passes secrets/config by rendering (rather than setting env vars), I went with the approach of explicitly passing them as arguments

### CODE REVIEW

**Comment by burnsed** [comment]
Should the scripts be more specifically named (i.e. `circleci-docker-publish` instead of `circle-docker`)? I'm just wondering if given circle's model we will ever have more than one docker, npm, or catapult script?

File: README.md [hunk 1]
```diff
@@ -1,2 +1,55 @@
...
+```
+$ ./circleci-docker <docker_user> <docker_pass> <docker_email> <org>
+```
+
+Publishes an image to DockerHub at `<org>/<repo>`.
+Tags the image according to the git short commit sha; this is 7 characters long and equal to `$(git rev-parse --short HEAD)`.
+
+### Npm
+
+_Not yet implemented._
```
Comment (nathanleiby) [inline README.md]: fixed labeling of NYI 	9ac7bdf

**Comment by nathanleiby** [comment]
@burnsed I agree with you -- I renamed docker to docker-publish and npm to npm-publish.. cbedf79

I also put them in a circleci folder to imply that they're written the circleci in mind specifically...  (rather than having that in the script name).. ultimately, if they're factored more generally we can keep them out of the folder

**Comment by burnsed** [comment]
LGTM!

**Pull request merged**